app/activeproject.cpp

```diff
@@ -483,31 +456,31 @@ void ActiveProject::setMapTheme( const QString &themeName )
 
   emit mapThemeChanged( mMapTheme );
 
-  updateRecordingLayers(); // <- worth to decouple similar to map themes model decoupling
   updateActiveLayer();
   updateMapSettingsLayers();
 }
 
 void ActiveProject::updateActiveLayer()
 {
-  if ( !layerVisible( mActiveLayer.layer() ) )
+  if ( !InputUtils::layerVisible( mActiveLayer.layer() ) )
   {
-    QgsMapLayer *defaultLayer = mRecordingLayerPM.layerFromLayerName( mAppSettings.defaultLayer() );
+    QgsMapLayer *defaultLayer = InputUtils::mapLayerFromName( mAppSettings.defaultLayer(), mQgsProject );
 
-    if ( !defaultLayer )
+    const QMap<QString, QgsMapLayer *> layers = mQgsProject->mapLayers();
+    for ( auto it = layers.cbegin(); it != layers.cend(); ++it )
     {
-      defaultLayer = mRecordingLayerPM.firstUsableLayer();
+      QgsMapLayer *layer = it.value();
+      if ( InputUtils::recordingAllowed( layer, mQgsProject ) )
+      {
+        defaultLayer = layer;
+        break;
+      }
     }
```

Comment:
Why do you read the default layer from the app settings if you do not use it then? If we have some layer in the app settings, we want to pick that one (ofc, only if it is visible and can be used for recording)! :)